clients/http_json_rpc_client.go

```diff
@@ -91,24 +110,17 @@ func NewGenericHttpJsonRpcClient(
 
 			client.batchRequests = make(map[interface{}]*batchRequest)
 		}
+
 		if jsonRpcCfg.EnableGzip != nil {
 			client.enableGzip = *jsonRpcCfg.EnableGzip
 		}
+
 		if jsonRpcCfg.Headers != nil {
 			client.headers = jsonRpcCfg.Headers
 		}
-	}
 
-	if util.IsTest() {
-		client.httpClient = &http.Client{}
-	} else {
-		client.httpClient = &http.Client{
-			Timeout: 60 * time.Second,
-			Transport: &http.Transport{
-				MaxIdleConns:        1024,
-				MaxIdleConnsPerHost: 256,
-				IdleConnTimeout:     90 * time.Second,
-			},
+		if jsonRpcCfg.ProxyPool != "" {
+			client.proxyPool = proxyPool
```

Comment:
```suggestion
		client.proxyPool = proxyPool
```
do we need this check?

Reply:
good point. the proxyPoolsRegistry is already setup at a higher level, so this check does not accomplish anything